usr/lib/common/loadsave.c

```diff
@@ -2634,15 +2648,31 @@ CK_RV reload_token_object(STDLL_TokData_t *tokdata, OBJECT *obj)
 
     set_perm(fileno(fp));
 
-    if (fread(header, HEADER_LEN, 1, fp) != 1) {
+    if (fread(header, 5, 1, fp) != 1) {
         OCK_SYSLOG(LOG_ERR, "Cannot read header\n");
         rc = CKR_FUNCTION_FAILED;
         goto done;
     }
 
     memcpy(&ver, header, 4);
     memcpy(&priv, header + 4, 1);
-    memcpy(&len, header + 60, 4);
+    if (priv) {
+        if (fread(header + 5, HEADER_LEN - 5, 1, fp) != 1) {
+            OCK_SYSLOG(LOG_ERR, "Cannot read header\n");
+            rc = CKR_FUNCTION_FAILED;
+            goto done;
+        }
+
+        memcpy(&len, header + 60, 4);
```

Comment:
Can we make this calculation a bit more self-explaining? Ok, 60 has been used before, but maybe you can also define a constant for this value as well.

Reply:
it would certainly make sense to use defines here  (or even better a struct and offsetof), but these offsets of the fields are used at many places within this source file, and we would need to change all of these places then. I would not want to do that now, but will put it on my list if things to do at a later time.